views/js/qtiCreator/widgets/interactions/matchInteraction/states/Map.js

```diff
@@ -4,44 +4,47 @@ define([
     'taoQtiItem/qtiCreator/widgets/states/Map',
     'taoQtiItem/qtiCreator/widgets/interactions/matchInteraction/ResponseWidget',
     'lodash'
-], function($, stateFactory, Map, responseWidget, _){
+], function($, stateFactory, Map, responseWidget, _) {
 
-    var MatchInteractionStateMap = stateFactory.create(Map, function(){
+    return stateFactory.create(Map, function() {
 
-        var _widget = this.widget,
+        const _widget = this.widget,
             interaction = _widget.element,
-            response = interaction.getResponseDeclaration();
+            response = interaction.getResponseDeclaration(),
+            matchInteractionArea = $('.match-interaction-area input[type="checkbox"]', _widget.$container);
 
         //init response widget in responseMapping mode:
         responseWidget.create(_widget, true);
 
         //finally, apply defined correct response and response mapping:
         responseWidget.setResponse(interaction, _.values(response.getCorrect()));
-        
-       //change the correct state 
-       _widget.on('metaChange', function(meta){
-            if(meta.key === 'defineCorrect'){
-                if(meta.value){
-                    $('.match-interaction-area input[type="checkbox"]', _widget.$container)
-                        .removeProp('disabled') 
+
+        matchInteractionArea
+            .prop('disabled', true)
+            .prop('checked', false)
+            .addClass('disabled');
+
+       //change the correct state
+        _widget.on('metaChange', function(meta) {
+            if (meta.key === 'defineCorrect') {
+                if (meta.value) {
+                    matchInteractionArea
+                        .removeProp('disabled')
                         .removeClass('disabled');
                 } else {
-                    $('.match-interaction-area input[type="checkbox"]', _widget.$container)
+                    matchInteractionArea
                         .prop('disabled', true)
                         .prop('checked', false)
                         .addClass('disabled');
-                }
-                if(meta.value === false){
                     response.setCorrect([]);
                 }
             }
         });
 
     }, function(){
-
+        $('.match-interaction-area input[type="checkbox"]', this.widget.$container)
+            .removeProp('disabled')
+            .removeClass('disabled');
         responseWidget.destroy(this.widget);
-
     });
-
-    return MatchInteractionStateMap;
 });
```

Comment:
Could you add new line, I don't like this icon :(

Reply:
Done